### PR TITLE
Disambiguate sub2ind

### PR DESCRIPTION
**Pull request by timholy**
Also adds a broken test for #18307

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -1609,7 +1609,10 @@ _div(ind, d::Integer) = div(ind, d), 1, d
 _div(ind, r::AbstractUnitRange) = (d = unsafe_length(r); (div(ind, d), first(r), d))
 
 # Vectorized forms
-function sub2ind{N,T<:Integer}(inds::Union{Dims{N},Indices{N}}, I::AbstractVector{T}...)
+sub2ind{T<:Integer}(inds::Indices{1}, I1::AbstractVector{T}, I::AbstractVector{T}...) = throw(ArgumentError("Linear indexing is not defined for one-dimensional arrays"))
```
Comment (tkelman) [inline base/abstractarray.jl]: wrap this?

**Comment by tkelman** [comment]
Does `@test_broken` work properly with a `@test_throws` ?

**Comment by timholy** [comment]
Seems to, at least in this direction.

**Comment by tkelman** [comment]
but will it serve the purpose of test_broken correctly? if something fixes the issue, will the test_broken flag it when it happens, or silently pass?

**Comment by vtjnash** [comment]
It'll silently pass, since `@test_throws` doesn't return true/false:
```
julia> Base.Test.@test_broken Base.Test.@test_throws AssertionError @assert true
Test Failed
  Expression: @assert true
    Expected: AssertionError
  No exception thrown
Test Broken
Expression: Base.Test.@test_throws AssertionError @assert(true)


julia> Base.Test.@test_broken Base.Test.@test_throws AssertionError @assert false
Test Broken
Expression: Base.Test.@test_throws AssertionError @assert(false)
```

**Comment by timholy** [comment]
Shall I just delete the test? Or comment it out?

**Comment by tkelman** [comment]
don't use test_broken, but do check the current result with a comment that it's currently wrong and should be fixed

**Comment by timholy** [comment]
OK, see if this is better.

**Pull request merged**